src/services/services.ts

```diff
@@ -2928,8 +2967,26 @@ namespace ts {
             log("getCompletionData: Is inside comment: " + (new Date().getTime() - start));
 
             if (insideComment) {
-                log("Returning an empty list because completion was inside a comment.");
-                return undefined;
+                // Completion should work inside certain JsDoc tags. For example:
+                //     /** @type {number | string} */
+                // Completion should work in the brackets
+                let insideJsDocTagExpression = false;
+                let tag = getJsDocTagAtPosition(sourceFile, position);
+                if (tag) {
+                    switch (tag.kind) {
+                        case SyntaxKind.JSDocTypeTag:
+                        case SyntaxKind.JSDocParameterTag:
+                            let tagWithExpression = <JSDocTypeTag | JSDocParameterTag>tag;
+                            if (tagWithExpression.typeExpression) {
+                                insideJsDocTagExpression = tagWithExpression.typeExpression.pos < position && position < tagWithExpression.typeExpression.end;
+                            };
+                            break;
+                    }
+                }
+                if (!insideJsDocTagExpression) {
+                    log("Returning an empty list because completion was inside a regular comment or plain text part of a JsDoc comment.");

```

Comment:
Right, though that is when ``insideJsDocTagExpression`` is true. The commented line is when the cursor is out of the braces.